lib/hammer_cli_foreman/template.rb

```diff
@@ -142,6 +142,36 @@ class DeleteCommand < HammerCLIForeman::DeleteCommand
       build_options
     end
 
+    class ExportCommand < HammerCLIForeman::DownloadCommand
+      command_name    "export"
+      action :export
+
+      def default_filename
+        "Template-#{Time.new.strftime("%Y-%m-%d")}.txt"
+      end
+
+      build_options
+
+      def execute
+        response = send_request
+        if response.code == 204
+          print_message(_('The template is not ready yet.'))
+          HammerCLI::EX_TEMPFAIL
+        else
+          handle_success(response)
+          HammerCLI::EX_OK
+        end
+      end
+
+      def handle_success(response)
+        if option_path
+          filepath = store_response(response)
+          print_message(_('The template has been saved to %{path}.'), {:path => filepath})
+        else
+          puts response.body
```

Comment:
This can be actually moved back into `execute` body. And as I see there is only one difference between this and the parent command: `response` vs `template` word, which makes me think that the superclass can be a bit refactored to allow redefining only the message to use in `print_message`...

Reply:
Maybe we should add an optional param to execute to indicate which template was saved. (In a different PR)

Reply:
So I guess this PR is now blocked untill we resolve https://github.com/theforeman/hammer-cli-foreman/pull/598.